python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -245,17 +276,15 @@ def runAssociation(self, sensorRef):
         self.log.info("Running Association...")
 
         diffType = self.config.differencer.coaddName
-        try:
-            catalog = sensorRef.get(diffType + "Diff_diaSrc")
-            exposure = sensorRef.get(diffType + "Diff_differenceExp")
-            result = self.associator.run(catalog, exposure)
-        finally:
-            # Stateful AssociationTask will work for now because TaskRunner
-            # uses task-oriented parallelism. Will not be necessary after DM-13672
-            self.associator.level1_db.close()
+
+        catalog = sensorRef.get(diffType + "Diff_diaSrc")
+        exposure = sensorRef.get(diffType + "Diff_differenceExp")
+
+        dia_sources = self.diaSourceDpddifier.run(catalog, exposure)
```

Comment:
I assume (hope) the Dpddifier is a temporary measure; is there some ticket we can `TODO` here?

Reply:
Not currently. Currently the DPDD-ification process Yusra is working on is still a prototype. At some point we may switch over to that but there will always be some process in between the output of ip_diffim and ap_association/pipe.